internal/plugin/connectors/tcp/mssql/mock/options.go

```diff
@@ -26,76 +26,43 @@ type MSSQLConnectorCtorOptions struct {
 	ServerLoginResponse *mssql.LoginResponse
 }
 
-// MSSQLConnectorCtorOption is the 'functional option' complement to
-// MSSQLConnectorCtorOptions.
-type MSSQLConnectorCtorOption func(*MSSQLConnectorCtorOptions)
-
-// MSSQLConnectorCtor is a 'functional option' of types.ConnectorOption. It generates
-// a customizable MSSQLConnectorCtor using MSSQLConnectorCtorOptions, then sets it on the
-// ConnectorOption passed to it.
-//
-// Empty values of MSSQLConnectorCtorOption result default:
-// ServerPreloginResponse => map[uint8][]byte{}
-// ServerLoginResponse => &mssql.LoginResponse{}
-// BackendConn => nil
-// BackendConn => nil
-// ClientLoginRequestPtr => nil (this means ClientLoginRequest is read from the channel to
-// a vacuum)
-func MSSQLConnectorCtor(setters ...MSSQLConnectorCtorOption) types.ConnectorOption {
-	args := &MSSQLConnectorCtorOptions{}
-
-	for _, setter := range setters {
-		setter(args)
-	}
+// NewMSSQLConnectorCtor is the default constructor for MSSQLConnectorCtor
+var NewMSSQLConnectorCtor = MSSQLConnectorCtor{
+	BackendConn:            NewNetConn(nil),
+	Err:                    nil,
+	ServerPreloginResponse: map[uint8][]byte{},
+	ClientLoginRequestPtr:  nil,
+	ServerLoginResponse:    &mssql.LoginResponse{},
+}
 
-	return func(connectorArgs *types.ConnectorOptions) {
-		connectorArgs.NewMSSQLConnector = NewSuccessfulMSSQLConnectorCtor(
+// DefaultConnectorOption is a 'functional option' of MSSQLConnectorCtor.
+// It generates a customizable MSSQLConnectorCtorOption using MSSQLConnectorCtor,
+// then sets it on the ConnectorOption passed to it, along with default Connector methods.
+func DefaultConnectorOption(ctor MSSQLConnectorCtor) types.ConnectorOption {
+	return func(connectOptions *types.ConnectorOptions) {
+		connectOptions.Logger = logmock.NewLogger()
+		connectOptions.NewMSSQLConnector = NewSuccessfulMSSQLConnectorCtor(
 			func(ctx context.Context) (net.Conn, error) {
 				interceptor := mssql.ConnectInterceptorFromContext(ctx)
 
-				if args.ServerPreloginResponse == nil {
-					args.ServerPreloginResponse = map[uint8][]byte{}
-				}
-				interceptor.ServerPreLoginResponse <- args.ServerPreloginResponse
+				interceptor.ServerPreLoginResponse <- ctor.ServerPreloginResponse
```

Comment:
How come we no longer have the nil protection for `ctor.ServerPreloginResponse` and `interceptor.ServerLoginResponse` ?

Reply:
It’s no longer needed, since the values there will never be nil. This method is never used for a fail case so that shouldn’t be necessary.